test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -4469,7 +4469,22 @@ TEST_P(SslSocketTest, OverrideApplicationProtocols) {
   // in the config.
   server_ctx->add_alpn_protocols("test");
   transport_socket_options = std::make_shared<Network::TransportSocketOptionsImpl>(
-      "", std::vector<std::string>{}, std::vector<std::string>{}, "test");
+      "", std::vector<std::string>{}, std::vector<std::string>{}, std::vector<std::string>{"test"});
+  testUtilV2(test_options.setExpectedALPNProtocol("test").setTransportSocketOptions(
+      transport_socket_options));
+
+  // With multiple matching fallbacks specified, a single match will match.
+  server_ctx->add_alpn_protocols("foo");
+  transport_socket_options = std::make_shared<Network::TransportSocketOptionsImpl>(
+      "", std::vector<std::string>{}, std::vector<std::string>{},
+      std::vector<std::string>{"foo", "test"});
+  testUtilV2(test_options.setExpectedALPNProtocol("test").setTransportSocketOptions(
+      transport_socket_options));
+
+  // With multiple fallbacks specified, a single match will match.
+  transport_socket_options = std::make_shared<Network::TransportSocketOptionsImpl>(
+      "", std::vector<std::string>{}, std::vector<std::string>{},
+      std::vector<std::string>{"foo", "test"});
```

Comment:
Are these two tests identical?

Reply:
Whoops, yes!  It was supposed to test
 foo vs foo, test 
then
foo, test vs foo, test
but I added the second match to the earlier not the later.  good spotting!